### DISCUSSION TITLE
Are there any issues with Generator of a certain class or I misunderstood it?

### DISCUSSION THREAD

**Discussion by szw0407** [microsoft/pylance-release#5167] [Q&A]
A brief example code:
```python
from typing import Generator

class a:
    def __init__(self, val: int):
        self.val = val
l: list[a] = [a(i) for i in range(10)]

def generate()-> Generator[a]: 
    # here is a type error.
    # Pylance says that the generator should be a generator of a | Any | None, but not a
    # However, the type of the generator should be a generator of instances of a.
    #  In no occasion should it be a generator of Any, is it?
    yield from l
```

Pylance is switched to strict mode, and I just could not understand what mistake I have made in such code.

I have searched for relavant discussions and issues but nothing helped. Please help me understand it better. Thanks.

**Comment by erictraut** [comment]
`Generator` is a generic class that requires three type arguments. You've supplied only one in your example above. The first type argument is the "yield type" (the type that the generator yields each iteration), the second is the "send type" (the type that the generator expects to be sent), and the third is the "return type" (the type that the generator returns when it's complete). In your example, the correct return type for `generate` is `Generator[a, Any, None]`.
